packages/eip-5792-middleware/src/hooks/processSendCalls.ts

```diff
@@ -410,6 +462,79 @@ function validateCapabilities(sendCalls: SendCallsPayload) {
       )}`,
     );
   }
+
+  if (capabilities?.auxiliaryFunds) {
+    validateAuxFundsSupportAndRequiredAssets({
+      auxiliaryFunds: capabilities.auxiliaryFunds,
+      chainId,
+      keyringType,
+      isAuxiliaryFundsSupported,
+    });
+  }
+}
+
+/**
+ * Validates EIP-7682 optional `requiredAssets` to see if the account and chain are supported, and that param is well-formed.
+ *
+ * docs: {@link https://eips.ethereum.org/EIPS/eip-7682#extended-usage-requiredassets-parameter}
+ *
+ * @param param - The parameter object.
+ * @param param.auxiliaryFunds - The auxiliaryFunds param to validate.
+ * @param param.auxiliaryFunds.optional - Metadata to signal for wallets that support this optional capability, while maintaining compatibility with wallets that do not.
+ * @param param.auxiliaryFunds.requiredAssets - Metadata that enables a wallets support for `auxiliaryFunds` capability.
+ * @param param.chainId - The chain ID of the incoming request.
+ * @param param.keyringType - The type of keyring associated with the account.
+ * @param param.isAuxiliaryFundsSupported - Function to validate if auxiliary funds capability is supported.
+ * @throws JsonRpcError if auxiliary funds capability is not supported.
+ */
+function validateAuxFundsSupportAndRequiredAssets({
+  auxiliaryFunds,
+  chainId,
+  keyringType,
+  isAuxiliaryFundsSupported,
+}: {
+  auxiliaryFunds: {
+    optional?: boolean;
+    requiredAssets?: SendCallsRequiredAssetsParam[];
+  };
+  chainId: Hex;
+  keyringType: KeyringTypes;
+  isAuxiliaryFundsSupported: (chainId: Hex) => boolean;
+}) {
+  // If we can make use of that capability then we should, but otherwise we can process the request and ignore the capability
+  // so if the capability is signaled as optional, no validation is required, so we don't block the transaction from happening.
+  if (auxiliaryFunds.optional) {
+    return;
+  }
+  const isSupportedAccount =
+    KEYRING_TYPES_SUPPORTING_7702.includes(keyringType);
+
+  if (!isSupportedAccount) {
+    throw new JsonRpcError(
+      EIP5792ErrorCode.UnsupportedNonOptionalCapability,
+      `Unsupported non-optional capability: ${SupportedCapabilities.AuxiliaryFunds}`,
+    );
+  }
+
+  if (!isAuxiliaryFundsSupported(chainId)) {
+    throw new JsonRpcError(
+      EIP7682ErrorCode.UnsupportedChain,
+      `The wallet no longer supports auxiliary funds on the requested chain: ${chainId}`,
+    );
+  }
+
+  if (!auxiliaryFunds?.requiredAssets) {
+    return;
+  }
+
+  for (const asset of auxiliaryFunds.requiredAssets) {
+    if (asset.standard !== 'erc20') {
```

Comment:
Do we have a `tokenStandard` enum we could use here?

Reply:
I see a few competing ones around various modules/repos 🤔 

Ok this is fine for now